elements/lisk-p2p/test/unit/peer/base.ts

```diff
@@ -269,65 +302,87 @@ describe('peer/base', () => {
 				procedure: 'myProcedure',
 			};
 
-			return expect(defaultPeer.request(p2pPacket)).to.be.rejectedWith(
+			return expect(defaultPeer.request(p2pPacket)).rejects.toThrow(
 				'Peer socket does not exist',
 			);
 		});
 
 		it('should emit if socket exists', () => {
+			// Arrange
 			const p2pPacket = {
 				data: 'myData',
 				procedure: 'myProcedure',
 			};
-			const socket = createSocketStubInstance();
-			(defaultPeer as any)._socket = socket;
+			(defaultPeer as any)._socket = createSocketStubInstance();
+
+			// Act
 			defaultPeer.request(p2pPacket);
-			expect(socket.emit).to.be.calledOnceWith(REMOTE_SC_EVENT_RPC_REQUEST, {
-				type: RCP_REQUEST_TYPE,
-				procedure: p2pPacket.procedure,
-				data: p2pPacket.data,
-			});
+
+			// Assert
+			expect((defaultPeer as any)._socket.emit).toHaveBeenCalledTimes(1);
+			expect((defaultPeer as any)._socket.emit).toBeCalledWith(
+				REMOTE_SC_EVENT_RPC_REQUEST,
+				{
+					type: RCP_REQUEST_TYPE,
+					procedure: p2pPacket.procedure,
+					data: p2pPacket.data,
+				},
+				expect.any(Function),
+			);
 		});
 	});
 
 	describe('#fetchPeers', () => {
 		it('should call request', async () => {
-			sandbox.stub(defaultPeer, 'request').resolves({
+			const peerRequest = jest.spyOn(defaultPeer, 'request').mockResolvedValue({
 				data: {
 					peers: [],
 					success: true,
 				},
 			});
+
 			await defaultPeer.fetchPeers();
-			expect(defaultPeer.request).to.be.calledOnceWith({
+
+			expect(peerRequest).toHaveBeenCalledTimes(1);
+			expect(peerRequest).toHaveBeenCalledWith({
 				procedure: REMOTE_EVENT_RPC_GET_PEERS_LIST,
 			});
 		});
 
 		describe('when request() fails', () => {
 			beforeEach(() => {
-				sandbox.stub(defaultPeer, 'emit');
-				sandbox.stub(defaultPeer, 'request').throws('Error');
+				jest
+					.spyOn(defaultPeer, 'request')
+					.mockRejectedValue(EVENT_FAILED_TO_FETCH_PEERS);
+
+				(defaultPeer as any).emit = jest.fn();
 			});
 
 			it(`should emit ${EVENT_FAILED_TO_FETCH_PEERS} event`, async () => {
-				await expect(defaultPeer.fetchPeers()).to.be.rejected;
-				expect(defaultPeer.emit).to.be.calledOnceWith(
-					EVENT_FAILED_TO_FETCH_PEERS,
-				);
+				try {
```

Comment:
I know this was tricky to avoid but any change to give it one more try at removing the `try/catch` ?